### DISCUSSION TITLE
Spectate UX Redesign

### DISCUSSION THREAD

**Discussion by TheFantasticLoki** [ppy/osu#17495] [Ideas] [answered]
I'm bringing this up due to how frequently I get complaints about the ux when people try to spectate me in lazer.
The current system for trying to spectate a player is ABOMINABLE. It's hard to find how to do it in the first place (it took ppy himself 4 tries to tell me the actual correct place to look when I asked how to spectate people), you can't filter or search for a specific player and even if you can find them in the currently playing list, the player cards move around so often by the time you click spectate on the person you want to watch the list refreshes and you're spectating a different person.

There should be a dedicated button to spectate a player on their profile page inside lazer and when right clicking their name in chat in my opinion. This would make is so you don't have to scroll down a massive list of players to find the person you actually want to spectate and not have to worry about clicking spectate before the list refreshes, not to mention it's far more intuitive this way.

**Comment by Theighlin** [comment, marked as answer]
Solutions proposed are a duplicate of #14273

As for the overlay, it's probably waiting for a redesign, and i think (hope?) it will become like stable's one where you can search all online players and filter them.

**Reply by bdach** [reply]
correct, current list of players to spectate hasn't seen much serious work at all since its first implementation. will improve in due course.
